text/0000-dynamic-import-hints.md

```diff
@@ -0,0 +1,264 @@
+---
+title: Dynamic Import Hints and Its Metadata
+status: DRAFTED
+created_at: 2020-03-16
+updated_at: 2020-03-22
+rfc: https://github.com/salesforce/lwc-rfcs/pull/27
+champion: Aliaksandr Papko (@apapko) | LWC Team
+implementation: https://git.soma.salesforce.com/lwc/lwc-api/pull/42 & https://git.soma.salesforce.com/lwc/lwc-platform/pull/361
+---
+
+# Hints Syntax and Its Metadata
+
+## Summary
+This RFC suggests authoring syntax for specifying dynamic import hints and describes its metadata shape.
+
+## Motivation
+Using hints allow component authors to declare the component preload context by parameterizing dynamic imports. The values from the hints can be used to optimize dependency resolution and pre-fetching.
+
+## Hint Invariants
+* Hint key and value are string literals
+* Hint keys and values must be statically verifiable i.e. no curly braces to implicitly refer to a variable and such
+* Multiple hints will be allowed but with restrictions
+
+## Platform Specific Hint Invariants
+For the salesforce platform, the following invariants are proposed
+* Hint values must be globally available and resolvable on the server side
```

Comment:
How will this be enforced?

Reply:
Will remove it. We can find another place to capture the salesforce specific parts of this RFC.